### PR TITLE
MDL-78129 Add power level endpoint for matrix state apis

### PR DESCRIPTION
**Pull request by safatshahin**
Add another state api for power level.

### CODE REVIEW

File: application/src/Controller/MatrixController.php [hunk 1]
```diff
@@ -310,6 +310,12 @@ public function roomState(string $serverID, string $roomID, string $eventType, R
                 return $check['message'];
             }
             $room->setAvatar($payload->url);
+        } elseif ($eventType == 'm.room.power_levels') {
+            // Update room avatar.
```
Comment (andrewnicols) [inline application/src/Controller/MatrixController.php]: Incorrect comment.

File: application/src/Controller/MatrixController.php [hunk 2]
```diff
@@ -310,6 +310,12 @@ public function roomState(string $serverID, string $roomID, string $eventType, R
...
+            $check = $this->validateRequest((array)$payload, ['users']);
+            if (!$check['status']) {
+                return $check['message'];
+            }
```
Comment (andrewnicols) [inline application/src/Controller/MatrixController.php:318]: There isn't any update call. I'd expect something here, or a comment to explain that we aren't doing it?

**Review by andrewnicols — CHANGES_REQUESTED** [review]

**Comment by safatshahin** [comment]
Updated the comment, it was copy paste error.
None of the state api's in mock is setting anything, they are just validating the request that the required data in the payload is available.

File: application/src/Controller/MatrixController.php (hunk 2 above)
Comment (safatshahin) [reply application/src/Controller/MatrixController.php:318]: None of the state api's in mock is setting anything, they are just validating the request that the required data in the payload is available. Updated the comment block to explain more.

**Pull request merged**